### ISSUE TITLE
Infinite loop caused by `Style/IfUnlessModifier`, `Layout/EndAlignment`, `Layout/IndentationWidth` cops

### ISSUE THREAD

**Issue by ydakuka** [rubocop/rubocop#14857]
## Steps to reproduce the problem

Run RuboCop with autocorrect `rubocop -A -c /dev/null` on the following code:

```ruby
def _admin_menu_draw(items)
  res = []
  res << "<ul class='treeview-menu'>"
  items.each_with_index do |item, index|
    res << "<li class='#{'xn-openable' if item.key?(:items)} item_#{index + 1} #{'active' if is_active_menu(item[:key])}' #{item[:datas]}>
          #{_admin_menu_draw(item[:items]) if item.key?(:items)}
        </li>"
  end
  res << '</ul>'
  res.join
end
```

## Expected behavior

It should not enter an infinite correction loop.

## Actual behavior

> Infinite loop detected in file.rb and caused by Style/IfUnlessModifier -> Style/IfUnlessModifier -> Layout/EndAlignment, Layout/IndentationWidth, Style/IfUnlessModifier

--------

## RuboCop version

```
$ bundle exec rubocop -V
1.84.1 (using Parser 3.3.10.0, rubocop-ast 1.49.0, analyzing as Ruby 2.7, running on ruby 3.4.8) [x86_64-linux]
  - rubocop-performance 1.26.1
  - rubocop-rake 0.7.1
  - rubocop-rspec 3.9.0
```

**Comment by koic** [comment]
The issue could not be reproduced with any of the following commands: `rubocop -a`, `rubocop -A`, `rubocop --only Style/IfUnlessModifier,Layout/EndAlignment,Layout/IndentationWidth -a`, or `rubocop --only Style/IfUnlessModifier,Layout/EndAlignment,Layout/IndentationWidth -A`.

If a specific `.rubocop.yml` configuration or additional command-line options are required to reproduce the issue, please provide them as well.

**Comment by ydakuka** [comment]
I ran it on RuboCop's `.rubocop.yml` file (https://github.com/rubocop/rubocop/blob/master/.rubocop.yml).

If you remove the following cop, you will get the error when running `rubocop -A`:

```yaml
Layout/LineLength:
  Max: 100
  AllowedPatterns:
    - !ruby/regexp /\A +(it|describe|context|shared_examples|include_examples|it_behaves_like) ["']/
```

Alternatively, you can simply use this `.rubocop.yml`:

```yaml
AllCops:
  NewCops: enable
  TargetRubyVersion: 2.7
  SuggestExtensions: false
```
